rfcs/single_test.md

```diff
@@ -0,0 +1,301 @@
+# RFC: `single_test()` opt-in for single-page tests
+
+## Summary
+
+Single-page tests are simple testharness.js tests that just do assertions and finally call `done()`. They reduce boilerplate, as no `test()` or `async_test()` is needed. Callbacks also need not be wrapped with `step_func()`, as any error simply fails the test. See [simple sync test](https://github.com/web-platform-tests/wpt/blob/ded00e006a083cacc108e8a1e92963fe7b15de4e/mediacapture-streams/MediaDevices-SecureContext.html) and [simple async test](https://github.com/web-platform-tests/wpt/blob/87d9cd16b9b4992649dc2ea1ecb3261f163e9184/html/webappapis/timers/negative-setinterval.html) for real examples.
+
+Unfortunately, the rules for single-page tests are subtle and many tests accidentally enter this mode. This leads to less consistent results cross-browser, see [FileAPI example](https://wpt.fyi/results/FileAPI/url/url-format.any.worker.html?run_id=312160003&run_id=306970005&run_id=321820002&run_id=319900004).
+
+Introduce `single_test()` as an explicit opt-in for single-page tests. Once added to the existing ~130 tests, remove the old ways of enabling the mode, which happens inadvertently for up to ~640 tests in some browsers.
+
+Example:
+```js
+single_test();
+assert_false(someCondition);
+doSomething();
+addEventListener('someevent', () => {
+  assert_true(someCondition);
+  done();
+});
+```
+
+The `single_test()` method will optionally take the same `name` and `properties` arguments as other test types.
+
+Acknowledgments:
+- jgraham [added single-page tests in 2014](https://github.com/w3c/testharness.js/pull/67)
+- zcorpan [proposed the `single_test()` opt-in in 2018](https://github.com/web-platform-tests/wpt/pull/11364)
```

Comment:
You may also want to reference [gh-10853](https://github.com/web-platform-tests/wpt/issues/10853) since that has a substantial amount of discussion.